contracts/crosschain/ERC7802Bridge.sol

```diff
@@ -0,0 +1,161 @@
+    }
+
+    // TODO: check that bridge includes an address ? prevent override ?
+    function registerRemote(address token, bytes memory bridge, bytes memory remote) public virtual {
+        if (token != msg.sender) _checkOwner();
+
+        (bytes2 chainType, bytes memory chainReference, ) = bridge.parseV1();
+        bytes memory chain = InteroperableAddress.formatV1(chainType, chainReference, "");
+
+        // Check that there is a gateway registered for that chain (revert is no gateway is registered)
+        getGateway(chain);
+
+        _remoteBridges[token][chain] = bridge;
+        _remoteTokens[token][chain] = remote;
+
+        emit RemoteBridgeRegistered(token, chain);
+    }
+
+    function send(
+        address token,
+        bytes memory to,
+        uint256 amount,
+        bytes[] memory attributes
+    ) public payable virtual returns (bytes32) {
+        _fetchTokens(token, msg.sender, amount);
+
+        // identify destination chain
+        (bytes2 chainType, bytes memory chainReference, bytes memory recipient) = to.parseV1();
+        bytes memory destChain = InteroperableAddress.formatV1(chainType, chainReference, "");
+
+        // get details for that bridge: gateway, remote bridge, remote token
+        address gateway = getGateway(destChain);
+        bytes memory bridge = getRemoteBridge(token, destChain);
+        bytes memory remote = getRemoteToken(token, destChain);
+
+        // prepare payload
+        bytes memory payload = abi.encode(
+            remote,
+            InteroperableAddress.formatEvmV1(block.chainid, msg.sender),
+            recipient,
+            amount
+        );
+
+        // send crosschain signal
+        bytes32 sendId = IERC7786GatewaySource(gateway).sendMessage{value: msg.value}(bridge, payload, attributes);
+        emit Sent(token, msg.sender, to, amount);
+
+        return sendId;
+    }
+
+    function executeMessage(
+        bytes32 receiveId,
+        bytes memory sender,
+        bytes memory payload,
+        bytes[] memory /*attributes*/
+    ) public payable virtual returns (bytes4) {
+        // prevent duplicate
+        require(!_processed.get(uint256(receiveId)), ERC7802BridgeDuplicate());
+        _processed.set(uint256(receiveId));
+
+        // parse payload
+        (bytes memory local, bytes memory from, bytes memory recipient, uint256 amount) = abi.decode(
+            payload,
+            (bytes, bytes, bytes, uint256)
+        );
+
+        // identify source chain and validate corresponding gateway
+        (bytes2 chainType, bytes memory chainReference, ) = from.parseV1();
+        bytes memory srcChain = InteroperableAddress.formatV1(chainType, chainReference, "");
+        require(msg.sender == getGateway(srcChain), ERC7802BridgeInvalidGateway());
+
+        // identify local token (requested for mint) and validate sender is the correct bridge
+        (, address token) = local.parseEvmV1(); // todo: check chainid ?
```

Comment:
I think we should indeed check `chainId`. One can register a bridge that may have the same address in another chain (perhaps different code), so `getRemoteBridge` will return true.